transcom/mymove
# Int b 19925 sorting branch param to uppercase

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A948237)

## Summary

When searching for moves based on branch of service, the wrong number of moves was being returned. Upon further inspection it was discovered that if the affiliator entry wasn't capitalized in service_members then the query wouldn't return the correct number of moves given that branch. To fix this we are allowing for case-insensitive querying of moves based on service branch.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. git fetch and git checkout the branch. 
2. change the afilliator column of one of the entries in service_members to not be all uppercase but still the same word (i.e. ARMY -> Army)
3. Sign into the office app as a TOO or TIO and search for moves based on the service branch of the database entry you changed. 
4. Ensure the entry that you changed in the database is properly queried for in the move search and the total number of moves under that service branch matches the amount in the database.

### Backend



## Screenshots
Affiliation case change:


![Screenshot 2024-06-05 at 4 45 37 AM](https://github.com/transcom/mymove/assets/167351500/ddcf39c4-b1dd-4e6b-a5a2-bcedb95f45fb)


### Before:


![Screenshot 2024-06-05 at 4 33 02 AM](https://github.com/transcom/mymove/assets/167351500/86a2fd64-3034-4968-af73-f85ef82052d3)
![Screenshot 2024-06-05 at 4 47 31 AM](https://github.com/transcom/mymove/assets/167351500/5d5e9520-555e-422a-951d-474a912b2113)



### After:

![Screenshot 2024-06-05 at 4 25 58 AM](https://github.com/transcom/mymove/assets/167351500/20a3a861-615e-41a1-8ce1-2150056ede1a)
![Screenshot 2024-06-05 at 4 49 45 AM](https://github.com/transcom/mymove/assets/167351500/665d3cd5-971e-4c76-93f2-5f1c0112fcf2)
